src/session/user.rs

```diff
@@ -22,10 +22,10 @@ mod imp {
     pub struct User {
         pub id: Cell<i64>,
         pub type_: RefCell<Option<BoxedUserType>>,
-        pub first_name: RefCell<String>,
-        pub last_name: RefCell<String>,
-        pub username: RefCell<String>,
-        pub phone_number: RefCell<String>,
+        pub first_name: RefCell<Option<String>>,
+        pub last_name: RefCell<Option<String>>,
+        pub username: RefCell<Option<String>>,
+        pub phone_number: RefCell<Option<String>>,
```

Comment:
What's the advantage of wrapping Strings in Options?

Reply:
Let's say that this makes more sense since we don't need having a default String value because it will be replaced anyway when constructing the GObject.

Reply:
I know that in the end doing String::new() is still not doing allocations, but using Options is more consistent with the other objects and we can also differenciate between unset property and empty string property.

Reply:
How are we going to handle other types? For example, if we want to be consistent, we are also forced to wrap primitive types like integers.

Reply:
So basically I'd like to follow the rule of:
- Primitive types => `Cell<>`
- Objects => `RefCell<Option<>>`

Reply:
> ... we don't need having a default String value ...

I was referring to that statement. If we don't need a default string value of "" then we probably also don't need a default integer value of `0`. :-)

Reply:
Well, yeah, that point actually makes less sense since glib now should (hopefully) cover us about setting that value in the constructor before using it around. So, the remaining point is the consistency :p

Reply:
And I am also not sure what's your definition of an Object. 
Are that heap allocated types? Or is it the size of a type?
For example, this works fine in a `Cell`
```rust
#[derive(Copy, Clone)]
struct Large([u8; 10000000]);

fn main() {
    let t = std::cell::Cell::new(Large([1; 10000000]));
}
```

Reply:
> And I am also not sure what's your definition of an Object.

I meant a more complex type than the primitive ones.